packages/components/src/components/text-field/text-field.tsx

```diff
@@ -142,6 +152,7 @@ export class TextField {
       this.status === 'error' ? { 'aria-invalid': true } : {};
     const helperTextId = `helper-message-${i}`;
     const ariaDescribedByAttr = { 'aria-describedBy': helperTextId };
+    const datalist = { 'list': 'data-list' };
```

Comment:
I would put the conditional check here and not below when we spread the variable. Something like this:

```tsx
const list = this.list ? { 'list': this.list } : null;

return <input {...list} />
```